splunk_topology/check.py

```diff
@@ -164,31 +164,40 @@ def _dispatch_and_await_search(self, instance, saved_searches):
-            for message in response['messages']:
-                if message['type'] != "FATAL" and message['type'] != "INFO":
-                    self.log.info("Received unhandled message, got: " + str(message))
-
-            count += len(response["results"])
-            # process components and relations
-            if saved_search.element_type == "component":
-                fail_count += self._extract_components(instance, response)
-            elif saved_search.element_type == "relation":
-                fail_count += self._extract_relations(instance, response)
-
-        self.log.debug(
-            "Saved search done: %s in time %d with results %d of which %d failed" % (saved_search.name, time.time() - start_time, count, fail_count))
-
-        if fail_count is not 0:
-            if (fail_count is not count) and (count is not 0):
-                msg = "The saved search '%s' contained %d incomplete %s records" % (saved_search.name, fail_count, saved_search.element_type)
-                self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.WARNING, tags=instance.tags, message=msg)
-                self.log.warn(msg)
-                return False
-            elif count is not 0:
-                raise CheckException("All result of saved search '%s' contained incomplete data" % saved_search.name)
+
+        try:
+            responses = self._search(search_id, saved_search, instance)
+
+            for response in responses:
+                for message in response['messages']:
+                    if message['type'] != "FATAL" and message['type'] != "INFO":
+                        self.log.info("Received unhandled message, got: " + str(message))
+
+                count += len(response["results"])
+                # process components and relations
+                if saved_search.element_type == "component":
+                    fail_count += self._extract_components(instance, response)
+                elif saved_search.element_type == "relation":
+                    fail_count += self._extract_relations(instance, response)
+
+            self.log.debug(
+                "Saved search done: %s in time %d with results %d of which %d failed" % (saved_search.name, time.time() - start_time, count, fail_count))
+
+            if fail_count is not 0:
+                if (fail_count is not count) and (count is not 0):
+                    msg = "The saved search '%s' contained %d incomplete %s records" % (saved_search.name, fail_count, saved_search.element_type)
+                    self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.WARNING, tags=instance.tags, message=msg)
+                    self.log.warn(msg)
+                    return False
+                elif count is not 0:
+                    raise CheckException("All result of saved search '%s' contained incomplete data" % saved_search.name)
+
+        except CheckException as e:
+            if not instance.splunk_ignore_saved_search_errors:
+                raise e
```

Comment:
It would be cool to add a log line here before reraising with info on which saved search the issue occurred, otherwise this is not clear in the final logging

Reply:
Done.